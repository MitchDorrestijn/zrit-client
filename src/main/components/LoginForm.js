/**
 * React related imports
 */
import React from 'react';
import {
  Button,
  Form,
  Label,
  Input
} from 'reactstrap';

/**
 * Other imports
 */
import Logo from '../assets/img/logo.gif';

/**
 * This component renders the the login form
 */
const LoginForm = () => {
  return (
    <div className="loginGrandParentContaniner">
      <div className="loginParentContainer">
        <Form>
          <header>
            <img className="mb-4" src={Logo} alt="logo Zorgrit"/>
          </header>
          <Label for="inputUserName" className="float-left">
            <b>Gebruikersnaam</b>
          </Label>
          <Input type="text" id="inputUserName" className="mb-3 " placeholder="Gebruikersnaam" autoFocus />
          <Label for="inputPassword" className="float-left">
            <b>Wachtwoord</b>
          </Label>
          <Input type="password" id="inputPassword" className="mb-4" placeholder="Wachtwoord"/>
          <Button color="primary" className="crud-btn" size="lg" type="submit">Inloggen</Button>
        </Form>
      </div>
    </div>
  );
}

export default LoginForm;
