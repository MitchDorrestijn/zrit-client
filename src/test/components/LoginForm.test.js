import React from 'react';
import {shallow} from 'enzyme';
import toJSON from 'enzyme-to-json';
import LoginForm from '../../main/components/LoginForm';

const wrapper = shallow(<LoginForm />);

test('<LoginForm /> should match snapshot', () => {
    expect(toJSON(wrapper)).toMatchSnapshot();
});

test('<LoginForm /> should render one <Form />', () => {
    expect(wrapper.find('Form').length).toBe(1);
});

test('<LoginForm /> should render two <div></div> elements', () => {
    expect(wrapper.find('div').length).toBe(2);
});

test('<LoginForm /> should render one <header></header>', () => {
    expect(wrapper.find('header').length).toBe(1);
});

test('<LoginForm /> should render one <img />', () => {
    expect(wrapper.find('img').length).toBe(1);
});

test('<LoginForm /> should render <Label /> two times', () => {
    expect(wrapper.find('Label').length).toBe(2);
});

test('<LoginForm /> should render <Input /> two times', () => {
    expect(wrapper.find('Input').length).toBe(2);
});

test('<LoginForm /> should render one <Button />', () => {
    expect(wrapper.find('Button').length).toBe(1);
});
