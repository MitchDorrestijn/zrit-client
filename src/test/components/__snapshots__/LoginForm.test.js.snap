// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginForm /> should match snapshot 1`] = `
<div
  className="loginGrandParentContaniner"
>
  <div
    className="loginParentContainer"
  >
    <Form
      tag="form"
    >
      <header>
        <img
          alt="logo Zorgrit"
          className="mb-4"
          src="logo.gif"
        />
      </header>
      <Label
        className="float-left"
        for="inputUserName"
        tag="label"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        <b>
          Gebruikersnaam
        </b>
      </Label>
      <Input
        autoFocus={true}
        className="mb-3 "
        id="inputUserName"
        placeholder="Gebruikersnaam"
        type="text"
      />
      <Label
        className="float-left"
        for="inputPassword"
        tag="label"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        <b>
          Wachtwoord
        </b>
      </Label>
      <Input
        className="mb-4"
        id="inputPassword"
        placeholder="Wachtwoord"
        type="password"
      />
      <Button
        className="crud-btn"
        color="primary"
        size="lg"
        tag="button"
        type="submit"
      >
        Inloggen
      </Button>
    </Form>
  </div>
</div>
`;
